[app] Android 7 以下の端末でアプリが起動できない

## 概要
Android 7 以下の端末でアプリ起動すると、
下記のログを出力し、クラッシュしてしまいます。

ログにある`java.time` は[Android 8.0 からサポートされるようになった](https://developer.android.com/about/versions/oreo/android-8.0?hl=ja#java) ので、desugar 等の設定を加え、対応できればと思います。

``` log
FATAL EXCEPTION: main
Process: jp.co.yumemi.android.codecheck.debug, PID: 8176
java.lang.NoClassDefFoundError: java.time.OffsetDateTime
	at libcore.reflect.InternalNames.getClass(InternalNames.java:55)
	at java.lang.Class.getDexCacheType(Class.java:2551)
	at java.lang.reflect.Method.getReturnType(Method.java:150)
	at java.lang.Class.getDeclaredMethods(Class.java:1811)
	at com.squareup.moshi.AdapterMethodsFactory.get(AdapterMethodsFactory.java:118)
	at com.squareup.moshi.Moshi$Builder.add(Moshi.java:224)
	at io.github.tshion.android.codecheck.github.webapi.GitHubWebApi.<init>(GitHubWebApi.kt:52)
	at io.github.tshion.android.codecheck.github.webapi.GitHubWebApi.<init>(GitHubWebApi.kt:38)
	at jp.co.yumemi.android.code_check.MainApplication.onCreate(MainApplication.kt:31)
	at jp.co.yumemi.android.code_check.DebugApplication.onCreate(DebugApplication.kt:42)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1024)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5403)
	at android.app.ActivityThread.-wrap2(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1545)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6119)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
Caused by: java.lang.ClassNotFoundException: Didn't find class "java.ime.OffsetDateTime" on path: DexPathList[[zip file "/data/app/jp.co.umemi.android.codecheck.debug-1/base.apk"],nativeLibraryDirectories=[/ata/app/jp.co.yumemi.android.codecheck.debug-1/lib/x86, /system/lib, /endor/lib]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:380)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	at libcore.reflect.InternalNames.getClass(InternalNames.java:53)
	at java.lang.Class.getDexCacheType(Class.java:2551) 
	at java.lang.reflect.Method.getReturnType(Method.java:150) 
	at java.lang.Class.getDeclaredMethods(Class.java:1811) 
	at com.squareup.moshi.AdapterMethodsFactory.get(AdapterMethodsFactory.java:118) 
	at com.squareup.moshi.Moshi$Builder.add(Moshi.java:224) 
	at io.github.tshion.android.codecheck.github.webapi.GitHubWebApi.<init>(GitHubWebApi.kt:52) 
	at io.github.tshion.android.codecheck.github.webapi.GitHubWebApi.<init>(GitHubWebApi.kt:38) 
	at jp.co.yumemi.android.code_check.MainApplication.onCreate(MainApplication.kt:31) 
	at jp.co.yumemi.android.code_check.DebugApplication.onCreate(DebugApplication.kt:42) 
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1024) 
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5403) 
	at android.app.ActivityThread.-wrap2(ActivityThread.java) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1545) 
	at android.os.Handler.dispatchMessage(Handler.java:102) 
	at android.os.Looper.loop(Looper.java:154) 
	at android.app.ActivityThread.main(ActivityThread.java:6119) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) 
```


## 作業内容
### 完了条件
* Android 7 以下でもアプリが起動できる

### 想定される範囲
* 現時点では不明

### 対応しないこと
* 現時点では特になし



## 備考
### 参考文献
* [Java 8+ API desugaring support (Android Gradle Plugin 4.0.0+)](https://developer.android.com/studio/write/java8-support?hl=en#library-desugaring)

--- a/app/README.md
+++ b/app/README.md
@@ -62,6 +62,7 @@ Git 管理から外しているため、試す際は追加でセットアップ�
             * ui-ktx ([Maven Google](https://mvnrepository.com/artifact/androidx.navigation/navigation-ui-ktx))
         * [recyclerview](https://developer.android.com/jetpack/androidx/releases/recyclerview) ([Maven Google](https://mvnrepository.com/artifact/androidx.recyclerview/recyclerview))
     * [COIL](https://github.com/coil-kt/coil) ([Maven Central](https://mvnrepository.com/artifact/io.coil-kt/coil))
+    * [desugar](https://github.com/google/desugar_jdk_libs) ([Maven Google](https://mvnrepository.com/artifact/com.android.tools/desugar_jdk_libs))
     * Firebase BOM ([Maven Google](https://mvnrepository.com/artifact/com.google.firebase/firebase-bom))
         * [Firebase Crashlytics SDK](https://github.com/firebase/firebase-android-sdk/tree/master/firebase-crashlytics)
     * Kotlin Coroutines BOM ([Maven Central](https://mvnrepository.com/artifact/org.jetbrains.kotlinx/kotlinx-coroutines-bom))

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -64,6 +64,7 @@ android {
         }
     }
     compileOptions {
+        coreLibraryDesugaringEnabled true
         sourceCompatibility JavaVersion.VERSION_17
         targetCompatibility JavaVersion.VERSION_17
     }
@@ -97,6 +98,9 @@ dependencies {
 
     // Coil
     implementation "${libraries.coil}"
+
+    // Desugar
+    coreLibraryDesugaring "${libraries.desugar}"
 
     // Firebase
     releaseImplementation platform("${libraries.firebase.bom}")

--- a/variables.gradle
+++ b/variables.gradle
@@ -51,6 +51,7 @@ ext {
                     core   : 'org.jetbrains.kotlinx:kotlinx-coroutines-core',
                     test   : 'org.jetbrains.kotlinx:kotlinx-coroutines-test',
             ],
+            desugar    : 'com.android.tools:desugar_jdk_libs:2.0.4',
             firebase   : [
                     bom        : 'com.google.firebase:firebase-bom:32.5.0',
                     crashlytics: 'com.google.firebase:firebase-crashlytics',

